data/transactions/logic/eval.go

```diff
@@ -4060,3 +4061,62 @@ func opBase64Decode(cx *EvalContext) {
 	}
 	cx.stack[last].Bytes, cx.err = base64Decode(cx.stack[last].Bytes, encoding)
 }
+
+func parseJSON(jsonText []byte) (interface{}, error) {
+	if !json.Valid(jsonText) {
+		return nil, fmt.Errorf("invalid json text %s", jsonText)
+	}
+	var parsed interface{}
+	err := json.Unmarshal(jsonText, &parsed)
```

Comment:
What do we gain with the first test?  Does Unmarshal not return a proper error if the json is invalid?

Reply:
If you use a JSON decoder with the `UseNumber()` option, you can avoid converting parsed numbers into float64. Instead, they will get decoded as a `json.Number`, which is really just the raw string holding the number as text. For example:

```go
decoder := json.NewDecoder(bytes.NewReader(jsonText))
decoder.UseNumber()

var parsed interface{}
err := decoder.Decode(&parsed)
```

From there you could call [`Int64`](https://pkg.go.dev/encoding/json#Number.Int64) to convert to an int64, or create a new `big.Int` and call [`UnmarshalText`](https://pkg.go.dev/math/big#Int.UnmarshalText) on it to convert directly to a big int. We would probably want the `big.Int` approach so that the entire uint64 range can be properly decoded.

Reply:
> What do we gain with the first test? Does Unmarshal not return a proper error if the json is invalid?
json.Valid returns false for an input like {"key0": "algo"}/*comment*/, but unmarshal doesn't return an error.